internal/extender/resource.go

```diff
@@ -294,15 +298,35 @@ func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v
 	return reservedDriverNode, outcome, err
 }
 
-// Sort nodes by available resources ascending, with RAM usage more important.
-func lessThan(left resources.Resources, right resources.Resources) bool {
+// Sort by available resources ascending, with RAM usage more important.
+func lessThan(left *resources.Resources, right *resources.Resources) bool {
 	var memoryCompared = left.Memory.Cmp(right.Memory)
 	if memoryCompared != 0 {
 		return memoryCompared == -1
 	}
 	return left.CPU.Cmp(right.CPU) == -1
 }
 
+func groupNodesByAZ(nodes []*v1.Node) (nodesByAZ map[string][]*v1.Node) {
```

Comment:
now we have non trivial logic for sorting, can we move all sorting related code to its own package?

Reply:
you can also remove the named return value and declare it inside the function body